audio/filter/af_scaletempo2_internals.c

```diff
@@ -63,138 +63,20 @@ static void zero_2d_partial(float **a, int x, int y)
     }
 }
 
-// Energies of sliding windows of channels are interleaved.
-// The number windows is |input_frames| - (|frames_per_window| - 1), hence,
-// the method assumes |energy| must be, at least, of size
-// (|input_frames| - (|frames_per_window| - 1)) * |channels|.
-static void multi_channel_moving_block_energies(
-    float **input, int input_frames, int channels,
-    int frames_per_block, float *energy)
-{
-    int num_blocks = input_frames - (frames_per_block - 1);
-
-    for (int k = 0; k < channels; ++k) {
-        const float* input_channel = input[k];
-
-        energy[k] = 0;
-
-        // First block of channel |k|.
-        for (int m = 0; m < frames_per_block; ++m) {
-            energy[k] += input_channel[m] * input_channel[m];
-        }
-
-        const float* slide_out = input_channel;
-        const float* slide_in = input_channel + frames_per_block;
-        for (int n = 1; n < num_blocks; ++n, ++slide_in, ++slide_out) {
-            energy[k + n * channels] = energy[k + (n - 1) * channels]
-                - *slide_out * *slide_out + *slide_in * *slide_in;
-        }
-    }
-}
-
 static float multi_channel_similarity_measure(
-    const float* dot_prod_a_b,
-    const float* energy_a, const float* energy_b,
-    int channels)
-{
-    const float epsilon = 1e-12f;
-    float similarity_measure = 0.0f;
-    for (int n = 0; n < channels; ++n) {
-        similarity_measure += dot_prod_a_b[n]
-            / sqrtf(energy_a[n] * energy_b[n] + epsilon);
+    float **a, float **b, int frame_offset_b,
+    int channels, int num_frames)
+{
+    float distance = 0;
+    for (int c = 0; c < channels ; c++) {
+        float *source = b[c];
+        float *target = a[c];
+        for (int i = 0; i < num_frames; i++)
+            distance += fabs(target[i] - source[frame_offset_b + i]);
     }
-    return similarity_measure;
+    return -distance;
 }
```

Comment:
Add `restrict ` where appropriate. This will allow auto vectorizer to optimize this code. Especially that hand crafted vectorized code below is removed.